src/Components/Aspire.Microsoft.Azure.Cosmos/AspireAzureCosmosDBExtensions.cs

```diff
@@ -108,6 +109,19 @@ public static class AspireAzureCosmosDBExtensions
         if (serviceKey is null)
         {
             builder.Services.AddSingleton(_ => ConfigureDb());
+
+            var csBuilder = new DbConnectionStringBuilder();
+            csBuilder.ConnectionString = settings.ConnectionString;
+
+            if (csBuilder.TryGetValue("Database", out var databaseName))
+            {
+                builder.Services.AddSingleton<Database>(sp =>
+                {
+                    var client = sp.GetRequiredService<CosmosClient>();
+                    var database = client.CreateDatabaseIfNotExistsAsync(databaseName.ToString()).Result.Database;
```

Comment:
```suggestion
                    var database = client.GetDatabase(databaseName);
```

Doing a blocking metadata async call sounds not ideal? Also, what if the app interacts with multiple databases?

Reply:
If an app interacts with multiple databases, then the developer using Aspire would probably have wired up their app model a little differently and this code wouldn't kick in.